### PR TITLE
Added validation for repo schema version.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
cc @jsancio

### CODE REVIEW

**Comment by BenWhitehead** [comment]
Looks great! Merging

File: repo/meta/index.json [hunk 1]
```diff
@@ -112,5 +112,5 @@
       }
     }
   ],
-  "version":"0.1.0-alpha"
+  "version":"1.0.0-rc1"
```
Comment (mgummelt) [inline repo/meta/index.json]: Isn't this file auto-generated?  If so, where is this version coming from?

Comment (mgummelt) [inline repo/meta/index.json]: It looks like there's a repo/meta/version.json file that's still 0.1.0.  Is that a mistake?

**Comment by mgummelt** [comment]
why push this into a non-master branch?

**Comment by BenWhitehead** [comment]
@mgummelt master has been replaced by `version-1.x` as the default branch.  The intention behind this is that anything in the `version-1.x` branch is compatible with the version 1 schemas.

When we move to version 2 schemas in the future those will be targeted at different branches. This is to allow the cli to validate the repo's version so that it can ensure that it will be able to perform the actions it knows about.

**Comment by mgummelt** [comment]
yea, my mistake for not reading the eng email.  thanks

**Pull request merged**